projects/kit/components/input-date-range/input-date-range.component.ts

```diff
@@ -392,4 +407,9 @@ export class TuiInputDateRangeComponent
             ? new TuiDayRange(clampedBottom.from, availableMax)
             : clampedBottom;
     }
+
+    @tuiPure
```

Comment:
I think `tuiPure` is excessive for simple operations, like string concatenation, math of boolean. This function takes about 1-2 nanoseconds on my machine, I don't think it's worth bothering with `tuiPure`. As a rule of thumb, we use memoization when working with arrays or objects.